Algorithm WorkBench (Questions 4 & 6)

Question 4 (Algorithm Workbench)

Below is question 4 and its code provided, as you can see it has two functions "change_us(a,b)" and "main()". They're set to 1 and 3.4. It should print "1 3.4". Then, "change_us(x,y) sets 'a' and 'b' to 0 which prints '0 0'. Since this doesnt change 'x' and 'y' in 'main', the second print still shows '1 3.4'; meaining the programs output is: '1 3.4', '0 0', '1 3.4'. 

In [1]:
def main():
    x = 1
    y = 3.4
    print(x, y)
    change_us(x, y)
    print(x, y)

def change_us(a, b):
    a = 0
    b = 0
    print(a, b)

main()


1 3.4
0 0
1 3.4


As we can see above our hypothesis is correct! 

Question 6 (Algorithm Workbench)

Below is one of the many ways to do it, all I did was just import the random module, and assign it the variable the question wanted me to give it 'rand'; then printed rand to get a random number.

In [2]:
import random  # Import the random module to use random number functions

# Generate a random integer between 1 and 100 (inclusive) and assign it to the variable 'rand'
rand = random.randint(1, 100)

# Print the generated random number (optional)
print(rand)

62


Above we got a random number of 62 from 1-100.

In [ ]:
Programming Excercises (Questions 1, 16, 17, & 9) 

Question 1 (Programming Excercises)

Below is the code/program that asks the user to enter the distance in kilometers, which then uses a function to convert the distance asked to miles. Below I randomly used a distnace of '500 kilometers' which, if doing the math correctly, should give me 310 miles with some change.   

In [3]:
# Define the conversion function
def kilometers_to_miles(kilometers):
    # Conversion formula
    return kilometers * 0.6214

# Ask the user to enter a distance in kilometers
kilometers = float(input("Enter distance in kilometers: "))

# Convert the distance to miles
miles = kilometers_to_miles(kilometers)

# Print the result
print(f"{kilometers} kilometers is equal to {miles} miles.")

Enter distance in kilometers:  500


500.0 kilometers is equal to 310.7 miles.


As you can see from above , we guessed exaclty what the output would be correctly. All it took was using a function to start off then the formula we were given and a variable to print. 

In [ ]:
Question 16 (Programming Excercises)

First we have to understand what a prime function is, which the textbook states is "a number that is only evenly divisible by itself and 1". So the number we get below should not be divisible by anything other then itself OR 1.

In [8]:
# Define the is_prime function to check if a number is prime
def is_prime(n):
    # Numbers less than or equal to 1 are not prime
    if n <= 1:
        return False
    # Check for factors from 2 up to the square root of n
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

# Prompt the user to enter a number
number = int(input("Enter a number: "))

# Check if the number is prime and display the result
if is_prime(number):
    print(f"{number} is a prime number.")
else:
    print(f"{number} is not a prime number.")

Enter a number:  5


5 is a prime number.


Above we can see the number we got is prime and cant be divided by anything else other than itself or 1. I couldve inputted any number, but I chose to PURPOSEFULLY chose a prime number to prove that the code works. 

Question 17 (Programming Excercises)

Going of number 16 from before like it says to do in the textbook we can get/grasp a further idea of what it wants the coder to do. Below should just be a list from 1-100 of prime numbers.

In [7]:
# Define the is_prime function to check if a number is prime
def is_prime(n):
    # Numbers less than or equal to 1 are not prime
    if n <= 1:
        return False
    # Check for factors from 2 up to the square root of n
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

# Loop through numbers from 1 to 100 and check if they are prime
for num in range(1, 101):
    # If the number is prime, print it
    if is_prime(num):
        print(num)

2
3
5
7
11
13
17
19
23
29
31
37
41
43
47
53
59
61
67
71
73
79
83
89
97


Above, its kind of like question 16; instead of asking and inputting a number we just get a list of ALL the prime number from 1-100. We can fact check using the simple math and the definition we were previously given for number 16. 

Question 9 (Programming Excercises) 

In [ ]:
Going back a little in terms of the questions, below we have code that should ask us for feet, in which the code written wil convert from feet to inches using a multiplier.

In [9]:
# Define the feet_to_inches function to convert feet to inches
def feet_to_inches(feet):
    # There are 12 inches in one foot
    return feet * 12

# Prompt the user to enter a number of feet
feet = float(input("Enter the number of feet: "))

# Convert the feet to inches using the feet_to_inches function
inches = feet_to_inches(feet)

# Display the result
print(f"{feet} feet is equal to {inches} inches.")

Enter the number of feet:  16


16.0 feet is equal to 192.0 inches.


Above, I used 16 feet to give a demonstration of how it would look in terms of inches which is 192 feet.